qml/ComboSlider.qml

```diff
@@ -0,0 +1,94 @@
+import QtQuick 2.0
+import QtQuick.Controls 2.0
+import QtQuick.Layouts 1.12
+
+/* Usage:
+ ComboSlider {
+        id: slider
+        // not needed, used if it exists.
+        defaultValue: "Default!" 
+        model: [strings | numbers]
+        onOptionChanged: {
+            console.log("Value " + value + " at key " + key)
+        }
+    }
+*/
+
+Slider {
+    id: control
+
+    property alias model : internalTicks.model
+    property var defaultValue : null
+
+    signal optionChanged(var key, var value)
+
+    from: 0
+    to: model ? model.length - (defaultValue ? 0 : 1)
+              : 0
+
+    stepSize: 1
+
+    onValueChanged: {
+        if(model.length === value) {
+            optionChanged(-1, defaultValue)
+        } else {
+            optionChanged(value, model[value])
+        }
+    }
+
+    background: Rectangle {
+          id: bg
+          x: control.leftPadding
+          y: control.topPadding + control.availableHeight / 2 - height / 2
+          implicitWidth: 200
+          implicitHeight: 4
+          width: control.availableWidth
+          height: implicitHeight
+          radius: 2
+          color: "#bdbebf"
```

Comment:
We should pull the color from style manager, I think. https://github.com/bluerobotics/ping-viewer/blob/f893d00f5cbd1380bf9f84a244bf3c3d9362aa28/qml/main.qml#L19